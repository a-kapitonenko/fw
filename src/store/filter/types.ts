import { Frame } from '../frames/types';

export type Field = {
  name: string;
  value: string;
  checked?: boolean;
};

export type Groups = {
  color: Field[];
  width: Field[];
  noseBridge: Field[];
  shape: Field[];
  material: Field[];
};

export const enum FilterActionTypes {
  FETCH_REQUEST = '@@filter/FETCH_REQUEST',
  FETCH_SUCCESS = '@@filter/FETCH_SUCCESS',
  FETCH_ERROR = '@@filter/FETCH_ERROR',
  CHANGE_CHECKED = '@@filter/CHANGE_CHECKED',
};

export interface IFilterState {
  readonly fetching: boolean;
  readonly groups: Groups;
  readonly errors: string;
  readonly query: Object;
  readonly frames: Frame[];
};
