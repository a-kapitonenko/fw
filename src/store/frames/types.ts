export type Frame = {
  upc: number;
  label: string;
  value: string; 
  img: string;
  compatibility?: boolean;
};

export const enum FramesActionTypes {  
  SUBMIT_START = '@@frames/SUBMIT_START',
  SUBMIT_SUCCESS = '@@frames/SUBMIT_SUCCESS',
  SUBMIT_FAILED = '@@frames/SUBMIT_FAILED',
  FETCH_SIMILAR_FRAMES_START = '@@frames/FETCH_SIMILAR_FRAMES_START',
  FETCH_SIMILAR_FRAMES_SUCCESS = '@@frames/FETCH_SIMILAR_FRAMES_SUCCESS',
  FETCH_SIMILAR_FRAMES_FAILED = '@@frames/FETCH_SIMILAR_FRAMES_FAILED',
  CHECK_FRAME_START = '@@frames/CHECK_FRAME_START',
  CHECK_FRAME_SUCCESS = '@@frames/CHECK_FRAME_SUCCESS',
  CHECK_FRAME_FAILED = '@@frames/CHECK_FRAME_FAILED',
  OPEN = '@@frames/OPEN',
  CLOSE = '@@frames/CLOSE',
  SET_SELECTED_FRAME = '@@frames/SET_SELECTED_FRAME',
  CLEAR_SELECTED_FRAME = '@@frame/CLEAR_SELECTED_FRAME',
  CLEAR_RESULT = '@@frames/CLEAR_RESULT',
};

export interface FramesState {
  readonly isFetching: boolean;
  readonly errors: string;
  readonly open: boolean;
  readonly selectedFrame: Frame;
  readonly similarFrames: {
    readonly isFetching: boolean;
    readonly errors: string;
    readonly data: Frame[];
  };
};
