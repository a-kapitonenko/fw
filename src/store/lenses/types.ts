export type Lens = {
  name: string;
  value: number;
};

export const enum LensesActionTypes {
  FETCH_LENSES_START = '@@lenses/FETCH_LENSES_START',
  FETCH_LENSES_SUCCESS = '@@lenses/FETCH_LENSES_SUCCESS',
  FETCH_LENSES_FAILED = '@@lenses/FETCH_LENSES_FAILED',

  SAVE_LENS_START = '@@lenses/SAVE_LENS_START',
  SAVE_LENS_SUCCESS = '@@lenses/SAVE_LENS_SUCCESS',
  SAVE_LENS_FAILED = '@@lenses/SAVE_LENS_FAILED',

  CHECK_LENS_START = '@@lenses/CHECK_LENS_START',
  CHECK_LENS_SUCCESS = '@@lenses/CHECK_LENS_SUCCESS',
  CHECK_LENS_FALIED = '@@lenses/CHECK_LENS_FAILED',
};

export interface ILensesState {
  readonly isFetching: boolean;
  readonly errors: string;
  readonly lenses: Lens[];
};
