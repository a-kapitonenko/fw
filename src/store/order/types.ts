import { Frame } from '../frames/types';
import { Lens } from '../lenses/types';

export type OculusInfo = {
  sphere: string;
  cyclinder: string;
  axis: string;
  addition: string;
  PDDistance: string;
  PDNear: string;
  prism: string;
};

export type Prescription = {
  OD: OculusInfo;
  OS: OculusInfo;
};

export type Blueprint = {
  img: string;
};

export type Barcode = {
  img: string;
}

export const enum OrderActionTypes {
  SET_RX_INFORMATION = '@@order/SET_RX_INFORMATION',
  SET_RECOMMENDATION = '@@order/SET_RECOMMENDATION',
  SET_LENS = '@@order/SET_LENS',
  SET_FRAME = '@@order/SET_FRAME',
  SET_MESSAGE = '@@order/SET_MESSAGE',
  SET_FITTING_PROPERTIES = '@@order/SET_FITTING_PROPERTIES',
  SET_FITTING_HEIGHT = '@@order/SET_FITTING_HEIGHT',
  SET_BLUEPRINT = '@@order/SET_BLUEPRINT',
  SET_BARCODE = '@@order/SET_BARCODE',
  SET_ERRORS = '@@order/SET_ERRORS',
  DELETE_ERRORS = '@@order/DELETE_ERRORS'
};

export interface IOrderState {
  readonly prescription: Prescription;
  readonly frame: Frame;
  readonly fittingHeight: number;
  readonly fittingProperties: any;
  readonly lens: Lens;
  readonly recommendation: string;
  readonly message: string;
  readonly blueprint: Blueprint;
  readonly barcode: Barcode;
  readonly errors: object;
};
