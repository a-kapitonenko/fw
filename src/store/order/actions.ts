import { action } from 'typesafe-actions';
import { ApplicationState } from '../index';
import { OrderActionTypes, Prescription, Blueprint, BossTypes, Boss } from './types';

export const submitStart = (boss: Boss) => action(OrderActionTypes.SUBMIT_START, boss);
export const submitSuccess = () => action(OrderActionTypes.SUBMIT_SUCCESS);
export const submitFailed = (message: string) => action(OrderActionTypes.SUBMIT_FAILED, message);

export const saveOrderStart = (state: ApplicationState) => action(OrderActionTypes.SAVE_ORDER_START, state);
export const saveOrderSuccess = () => action(OrderActionTypes.SAVE_ORDER_SUCCESS);
export const saveOrderFailed = (message: string) => action(OrderActionTypes.SAVE_ORDER_FAILED, message);

export const fetchOrderValuesStart = (id: string) => action(OrderActionTypes.FETCH_ORDER_VALUES_START, id);
export const fetchOrderValuesSuccess = (values: any) => action(OrderActionTypes.FETCH_ORDER_VALUES_SUCCESS, values);
export const fetchOrderValuesFailed = (message: string) => action(OrderActionTypes.FETCH_ORDER_VALUES_FAILED, message);

export const savePrescriptionStart = (prescription: Prescription) => action(OrderActionTypes.SAVE_PRESCRIPTION_START, prescription);
export const savePrescriptionSuccess = () => action(OrderActionTypes.SAVE_PRESCRIPTION_SUCCESS);
export const savePrescriptionFailed = (message: string) => action(OrderActionTypes.SAVE_PRESCRIPTION_FAILED, message);

export const saveFittingHeightStart = (boss: Boss, height: number) => action(OrderActionTypes.SAVE_FITTING_HEIGHT_START, { boss, height });
export const saveFittingHeightSuccess = (blueprint: Blueprint) => action(OrderActionTypes.SAVE_FITTING_HEIGHT_SUCCESS, blueprint);
export const saveFittingHeightFailed = (message: string) => action(OrderActionTypes.SAVE_FITTING_HEIGHT_FAILED, message);

export const setRxInformation = (type: string, field: string, value: string) => action(OrderActionTypes.SET_RX_INFORMATION, { type, field, value });
export const setRecommendation = (recommendation: string) => action(OrderActionTypes.SET_RECOMMENDATION, recommendation);
export const setMessage = (message: string) => action(OrderActionTypes.SET_MESSAGE, message);
export const setFittingProperties = (properties: any) => action(OrderActionTypes.SET_FITTING_PROPERTIES, properties);
export const setBoss = (type: BossTypes, value: any) => action(OrderActionTypes.SET_BOSS, { type, value });
export const disableRedirect = () => action(OrderActionTypes.DISABLE_REDIRECT);
