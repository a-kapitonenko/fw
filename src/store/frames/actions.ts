import { action } from 'typesafe-actions';
import { FramesActionTypes, Frame } from './types';
import { Boss } from '../order/types';

export const open = () => action(FramesActionTypes.OPEN);
export const close = () => action(FramesActionTypes.CLOSE);

export const submitStart = (boss: Boss, frame: Frame) => action(FramesActionTypes.SUBMIT_START, { boss, frame });
export const submitSuccess = () => action(FramesActionTypes.SUBMIT_SUCCESS);
export const submitFailed = (message: string) => action(FramesActionTypes.SUBMIT_FAILED, message);

export const fetchSimilarFramesStart = (boss: Boss) => action(FramesActionTypes.FETCH_SIMILAR_FRAMES_START, boss);
export const fetchSimilarFramesSuccess = (frames: Frame[]) => action(FramesActionTypes.FETCH_SIMILAR_FRAMES_SUCCESS, frames);
export const fetchSimilarFramesFailed = (message: string) => action(FramesActionTypes.FETCH_SIMILAR_FRAMES_FAILED, message);

export const checkFrameStart = (boss: Boss) => action(FramesActionTypes.CHECK_FRAME_START, boss);
export const checkFrameSuccess = () => action(FramesActionTypes.CHECK_FRAME_SUCCESS);
export const checkFrameFailed = (message: string) => action(FramesActionTypes.CHECK_FRAME_FAILED, message);

export const setSelectedFrame = (frame: Frame) => action(FramesActionTypes.SET_SELECTED_FRAME, frame);
export const clearSelectedFrame = () => action(FramesActionTypes.CLEAR_SELECTED_FRAME);

export const clearResult = () => action(FramesActionTypes.CLEAR_RESULT);
