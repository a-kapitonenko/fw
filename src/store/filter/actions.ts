import { action } from 'typesafe-actions';
import { FilterActionTypes, Groups } from './types';
import { Boss } from '../order/types';
import { Frame } from '../frames/types';

export const filteringStart = (boss: Boss, query: object) => action(FilterActionTypes.FILTERING_START, { boss, query });
export const filteringSuccess = (frames: Frame[]) => action(FilterActionTypes.FILTERING_SUCCESS, frames);
export const filteringFailed = (message: string) => action(FilterActionTypes.FILTERING_FAILED, message);

export const fetchGroupsStart = () => action(FilterActionTypes.FETCH_GROUPS_START);
export const fetchGroupsSuccess = (groups: Groups) => action(FilterActionTypes.FETCH_GROUPS_SUCCESS, groups);
export const fetchGroupsFailed = (message: string) => action(FilterActionTypes.FETCH_GROUPS_FAILED, message);

export const changeChecked = (type: string, name: string, value: boolean) => action(FilterActionTypes.CHANGE_CHECKED, { type, name, value });
export const clearChecked = () => action(FilterActionTypes.CLEAR_CHECKED);
export const addQuery = (type: string, value: any) => action(FilterActionTypes.ADD_QUERY, { type, value });
export const deleteQuery = (type: string, value: any) => action(FilterActionTypes.DELETE_QUERY, { type, value });
export const clearQuery = () => action(FilterActionTypes.CLEAR_QUERY);
export const clearResult = () => action(FilterActionTypes.CLEAR_RESULT);
