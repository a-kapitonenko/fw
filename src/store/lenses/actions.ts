import { Dispatch } from 'redux';
import { action } from 'typesafe-actions';

import { LensesActionTypes, Lens } from './types';
import { IOrderState, Prescription, BossTypes } from '../order/types';
import * as orderActions from '../order/actions';

import { getLenses } from '../../test/lenses';
import { saveLensInformation } from '../../test/order';

// export const fetchRequest = () => action(LensesActionTypes.FETCH_REQUEST);
// export const closeReauest = () => action(LensesActionTypes.CLOSE_REQUEST);
export const setErrors = (errors: string) => action(LensesActionTypes.SET_ERRORS, errors);
export const clearErrors = () => action(LensesActionTypes.CLEAR_ERRORS);
export const setLenses = (lenses: Lens[]) => action(LensesActionTypes.SET_LENSES, lenses);

export const fetchLenses: any = (order: IOrderState) => (dispatch: Dispatch) => {
  dispatch(orderActions.fetchRequest());

  return new Promise((resolver) => {
    const response = getLenses(order);
    
    setTimeout(() => {
      resolver(response);
    }, 1000);
  })
  .then((response: any) => {
    dispatch(orderActions.closeRequest());

    if (response.success) {
      dispatch(setLenses(response.lenses));
    } else { 
      dispatch(setErrors(response.errors));
    }
  })
  .catch((errors: any) => dispatch(setErrors(errors)))
};

export const saveLens: any = (prescription: Prescription, lens: Lens) => (dispatch: Dispatch) => {
  dispatch(orderActions.fetchRequest());

  return new Promise((resolver) => {
    const response = saveLensInformation(prescription, lens)

    setTimeout(() => {
      resolver(response);
    }, 1000);
  })
  .then((response: any) => {
    dispatch(orderActions.closeRequest());

    if (response.success) {
      dispatch(orderActions.setRecommendation(response.recommendation));
      dispatch(orderActions.setBoss(BossTypes.LENS, lens));

    } else { 
      dispatch(setErrors(response.errors));
    }
  })
  .catch((errors: any) => dispatch(setErrors(errors)))
};
