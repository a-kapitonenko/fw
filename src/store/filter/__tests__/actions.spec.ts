import { Groups } from '../types';
import { Boss } from '../../order/types';
import { Frame } from '../../frames/types';
import * as filterActions from '../actions';
import * as mockData from '../../../mockData';

describe('filter actions', () => {
  it('test filteringStart action creator', () => {
    const boss: Boss = mockData.boss;
    const query: object = {};

    expect(filterActions.filteringStart(boss, query)).toMatchSnapshot();
  });

  it('test filteringSuccess action creator', () => {
    const frames: Frame[] = mockData.frames;

    expect(filterActions.filteringSuccess(frames)).toMatchSnapshot();
  });

  it('test filteringFailed action creator', () => {
    const message: string = mockData.errorMessage;

    expect(filterActions.filteringFailed(message)).toMatchSnapshot();
  });

  it('test fetchGroupsStart action creator', () => {
    expect(filterActions.fetchGroupsStart()).toMatchSnapshot();
  });

  it('test fetchGroupsSuccess action creator', () => {
    const groups: Groups = mockData.groups;

    expect(filterActions.fetchGroupsSuccess(groups)).toMatchSnapshot();
  });

  it('test fetchGroupsFailed action creator', () => {
    expect(filterActions.fetchGroupsFailed(mockData.errorMessage)).toMatchSnapshot();
  });

  it('test changeChecked action creator', () => {
    const type: string = 'color';
    const name: string = 'Black';
    const value: boolean = true;

    expect(filterActions.changeChecked(type, name, value)).toMatchSnapshot();
  });

  it('test clearChecked action creator', () => {
    expect(filterActions.clearChecked()).toMatchSnapshot();
  });

  it('test addQuery action creator', () => {
    const type: string = 'color';
    const value: any = 'test';

    expect(filterActions.addQuery(type, value)).toMatchSnapshot();
  });

  it('test deleteQuery action creator', () => {
    const type: string = 'color';
    const value: any = 'test';

    expect(filterActions.deleteQuery(type, value)).toMatchSnapshot();
  });

  it('test clearQuery action creator', () => {
    expect(filterActions.clearQuery()).toMatchSnapshot();
  });

  it('test clearResult action creator', () => {
    expect(filterActions.clearResult()).toMatchSnapshot();
  });
});
