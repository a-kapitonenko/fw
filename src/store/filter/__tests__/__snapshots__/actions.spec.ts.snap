// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter actions test addQuery action creator 1`] = `
Object {
  "meta": undefined,
  "payload": Object {
    "type": "color",
    "value": "test",
  },
  "type": "@@filter/ADD_QUERY",
}
`;

exports[`filter actions test changeChecked action creator 1`] = `
Object {
  "meta": undefined,
  "payload": Object {
    "name": "Black",
    "type": "color",
    "value": true,
  },
  "type": "@@filter/CHANGE_CHECKED",
}
`;

exports[`filter actions test clearChecked action creator 1`] = `
Object {
  "meta": undefined,
  "payload": undefined,
  "type": "@@filter/CLEAR_CHECKED",
}
`;

exports[`filter actions test clearQuery action creator 1`] = `
Object {
  "meta": undefined,
  "payload": undefined,
  "type": "@@filter/CLEAR_QUERY",
}
`;

exports[`filter actions test clearResult action creator 1`] = `
Object {
  "meta": undefined,
  "payload": undefined,
  "type": "@@filter/CLEAR_RESULT",
}
`;

exports[`filter actions test deleteQuery action creator 1`] = `
Object {
  "meta": undefined,
  "payload": Object {
    "type": "color",
    "value": "test",
  },
  "type": "@@filter/DELETE_QUERY",
}
`;

exports[`filter actions test fetchGroupsFailed action creator 1`] = `
Object {
  "meta": undefined,
  "payload": "Error message",
  "type": "@@filter/FETCHING_GROUPS_FAILED",
}
`;

exports[`filter actions test fetchGroupsStart action creator 1`] = `
Object {
  "meta": undefined,
  "payload": undefined,
  "type": "@@filter/FETCHING_GROUPS_START",
}
`;

exports[`filter actions test fetchGroupsSuccess action creator 1`] = `
Object {
  "meta": undefined,
  "payload": Object {
    "color": Array [
      Object {
        "name": "Black",
        "value": "black",
      },
    ],
    "material": Array [
      Object {
        "name": "Acetate",
        "value": "acetate",
      },
    ],
    "noseBridge": Array [
      Object {
        "name": "Standard",
        "value": "standard",
      },
    ],
    "shape": Array [
      Object {
        "name": "Square",
        "value": "square",
      },
    ],
    "width": Array [
      Object {
        "name": "Narrow",
        "value": "narrow",
      },
    ],
  },
  "type": "@@filter/FETCHING_GROUPS_SUCCESS",
}
`;

exports[`filter actions test filteringFailed action creator 1`] = `
Object {
  "meta": undefined,
  "payload": "Error message",
  "type": "@@filter/FILTERING_FAILDE",
}
`;

exports[`filter actions test filteringStart action creator 1`] = `
Object {
  "meta": undefined,
  "payload": Object {
    "boss": Object {},
    "query": Object {},
  },
  "type": "@@filter/FILTERING_START",
}
`;

exports[`filter actions test filteringSuccess action creator 1`] = `
Object {
  "meta": undefined,
  "payload": Array [
    Object {
      "compatibility": true,
      "img": "1.png",
      "label": "FM17116",
      "upc": 82850553738,
      "value": "82850553738",
    },
    Object {
      "compatibility": false,
      "img": "2.png",
      "label": "SA17346",
      "upc": 82850523999,
      "value": "82850523999",
    },
  ],
  "type": "@@filter/FILTERING_SUCCESS",
}
`;
